robot/packages/deferred_package_manager.go

```diff
@@ -121,7 +120,7 @@ func (m *deferredPackageManager) getManagerForSync(ctx context.Context, packages
 	}
 
 	// if we are missing packages, run createCloudManager synchronously
```

Comment:
not actionable in this PR, but this feels like a case that can break with spotty connections

Reply:
@abe-winter yeah I'm curious about the scenario you're thinking of. But also agree that we can address as a follow-up.

Reply:
I'll admit I don't know what timeouts are operative here. From reading, I think this is called in the RobotFromConfig -> New path in local_robot.go.

worried about the case where we have 3 good packages + 1 bad package (added while on bad internet, for example, and failed to download, and then the power failed so the device restarted).

will `m.createCloudManager` block a long time in this case, i.e. will the 1 bad module prevent the 3 good ones from starting?

Reply:
Got it I see what you mean. Hmm I'm having issues finding the timeout. I looked up the stack and cannot see where in the context it is set before passing it down. May have missed it or is not set. It is probably this way on purpose so that slow connections can still download a longer time is needed.

I think the TODOs in the existing code point to this needing to be addressed in a different way: https://github.com/viamrobotics/rdk/blob/main/robot/impl/local_robot.go#L1119. Refactoring how package manager and module manager interact with each other would be the best approach IMO.

Also not sure if there is a huge difference in 3/4 of the modules on a machine working. Really depends on the usecase. I'd imagine a machine is mostly useless if one of the modules is not working. So this likely hasn't come up for this reason.

Reply:
Going to merge this in for now and we can keep talking about this.